custom/enikshay/integrations/ninetyninedots/repeater_generators.py

```diff
@@ -218,3 +218,52 @@ def handle_failure(self, response, episode_case, repeat_record):
                     ),
                 }
             )
+
+
+class UnenrollPatientPayloadGenerator(NinetyNineDotsBasePayloadGenerator):
+    deprecated_format_names = ('case_json',)
+
+    def get_payload(self, repeat_record, episode_case):
+        domain = episode_case.domain
+        person_case = get_person_case_from_occurrence(
+            domain, get_occurrence_case_from_episode(
+                domain, episode_case.case_id
+            ).case_id
+        )
+
+        if episode_case.closed:
+            reason = episode_case.get_case_property('close_reason')
+        elif episode_case.get_case_property('dots_99_enabled') == 'false':
+            reason = 'source_changed'
+        else:
+            reason = 'unknown'
+
+        payload = {
+            'beneficiary_id': person_case.case_id,
+            'reason': reason,
+        }
+        return json.dumps(payload)
+
+    def handle_success(self, response, episode_case, repeat_record):
+        if response.status_code == 200:
+            update_case(
+                episode_case.domain,
+                episode_case.case_id,
+                {
+                    "dots_99_registered": "false",
```

Comment:
Do we want to differentiate patients that
1. failed dots 99 notification.
2. unenrolled notification